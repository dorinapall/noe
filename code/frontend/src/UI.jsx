import React from "react";
import classNames from "classnames";
import { Link } from "react-router-dom";

export function View({ children }) {
  return (
    <div className="ViewContainer">
      <div className="View">{children}</div>
    </div>
  );
}

export function Caption({ children, ...props }) {
  const classes = classNames("Caption", "Text", { CenterText: props.center });
  return <h2 className={classes}>{children}</h2>;
}

export function Form({ children, onSubmit }) {
  return (
    <form className="Form" onSubmit={onSubmit}>
      {children}
    </form>
  );
}

export function Field({
  value,
  name,
  label,
  type,
  register,
  errors: allErrors,
  options,
  hidden,
}) {
  const errors = allErrors[name];
  return (
    <InputGroup hidden={hidden}>
      {type === "checkbox" ? (
        <Label htmlFor={name}>
          <Input
            value={value}
            name={name}
            label={label}
            type={type}
            register={register}
            options={options}
            id={name}
          />
          {label}
        </Label>
      ) : (
        <>
          {label && <Label htmlFor={name}>{label}</Label>}
          <Input
            value={value}
            name={name}
            label={label}
            type={type}
            register={register}
            options={options}
            id={name}
          />
        </>
      )}
      {errors && <HelpBlock error>{errors.message}</HelpBlock>}
    </InputGroup>
  );
}

export function Input({ value, name, type, register, options }) {
  if (type === "checkbox") {
    return (
      <input
        id={name}
        className="Input Inline"
        name={name}
        ref={register()}
        type={type}
        value={value} // Value instead of true
      />
    );
  } else if (type === "select") {
    return (
      <select className="Input" name={name} ref={register()} type={type}>
        {options.map((option) => (
          <option key={option.value} value={option.value}>
            {option.text}
          </option>
        ))}
      </select>
    );
  }
  return (
    <input
      className="Input"
      name={name}
      ref={register()}
      type={type || "text"}
    />
  );
}

export function InputGroup({ children, hidden }) {
  const classes = classNames("InputGroup", { Hidden: hidden });
  return <div className={classes}>{children}</div>;
}

export function Label({ children, className, htmlFor }) {
  return (
    <label className={`Label ${className || ""}`} htmlFor={htmlFor}>
      {children}
    </label>
  );
}

export function Button({
  children,
  type,
  onClick,
  toCenter,
  inverse,
  disabled,
}) {
  const classes = classNames("Button", {
    ToCenter: toCenter,
    Inverse: inverse,
    Disabled: disabled,
  });
  return (
    <button
      className={classes}
      type={type || "button"}
      onClick={onClick}
      disabled={disabled}
    >
      {children}
    </button>
  );
}

export function HelpBlock({ children, error }) {
  return <p>{children}</p>;
}

export function Text({ children, center, highlight, toCenter, light, strong }) {
  const classes = classNames("Text", {
    CenterText: center,
    HighlightText: highlight,
    ToCenter: toCenter,
    Light: light,
    Strong: strong,
  });
  return <p className={classes}>{children}</p>;
}

export function HighlightText({ children, center, ...props }) {
  const classes = classNames("Text", { CenterText: center });
  return (
    <Text highlight {...props}>
      {children}
    </Text>
  );
}

export function Image({ src }) {
  return (
    <div className="ImageContainer">
      <img src={src} />
    </div>
  );
}

export function LinkButton({ to, toCenter, inverse, children }) {
  const classes = classNames("Button", {
    ToCenter: toCenter,
    Inverse: inverse,
  });
  return (
    <Link className={classes} to={to}>
      {children}
    </Link>
  );
}

export function DataRow({ children }) {
  return <div className="DataRow">{children}</div>;
}

export function HR() {
  return <hr className="HR" />;
}
