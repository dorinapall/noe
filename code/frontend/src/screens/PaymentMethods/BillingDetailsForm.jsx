import React from "react";
import { useForm } from "react-hook-form";
import {
  Form,
  Field,
  InputGroup,
  Label,
  Flex,
  Input,
  HelpBlock,
  NextButton,
} from "../../UI";

export default function BillingDetailsForm({ onSubmit }) {
  const { register, handleSubmit, setError, errors } = useForm();
  const managedSubmit = handleSubmit((values) => onSubmit(values, setError));

  return (
    <Form onSubmit={managedSubmit}>
      <Field
        register={register}
        name="company_name"
        label="Név"
        placeholder="Példa Kft."
        errors={errors}
        required
      />
      <Field
        register={register}
        name="country"
        label="Ország"
        placeholder="Magyarország"
        errors={errors}
        required
      />
      <InputGroup
        hasError={errors.post_code || errors.city || errors.address_line1}
      >
        <Label>Tartózkodási cím</Label>
        <InputGroup>
          <Flex>
            <div style={{ flex: 2 }}>
              <Input
                register={register}
                name="post_code"
                placeholder="Irányítósz."
                required
              />
              {errors.post_code && (
                <HelpBlock>
                  {errors.post_code.type === "required"
                    ? "Ez a mező kötelező."
                    : errors.post_code.message}
                </HelpBlock>
              )}
            </div>
            <div style={{ flex: 5, marginLeft: 10 }}>
              <Input
                register={register}
                name="city"
                placeholder="Település"
                required
              />
              {errors.city && (
                <HelpBlock>
                  {errors.city.type === "required"
                    ? "Ez a mező kötelező."
                    : errors.city.message}
                </HelpBlock>
              )}
            </div>
          </Flex>
        </InputGroup>
        <Input
          register={register}
          name="address_line1"
          placeholder="Utca és Házszám"
          required
        />
        {errors.address_line1 && (
          <HelpBlock>
            {errors.address_line1.type === "required"
              ? "Ez a mező kötelező."
              : errors.address_line1.message}
          </HelpBlock>
        )}
      </InputGroup>
      <Field
        register={register}
        name="tax_number"
        label="Adószám"
        placeholder="123456789"
        errors={errors}
      />
      <NextButton type="submit">Véglegesítés</NextButton>
    </Form>
  );
}
