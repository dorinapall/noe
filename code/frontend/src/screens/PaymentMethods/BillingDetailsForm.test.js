import React from "react";
import { shallow } from "enzyme";
import { act } from "react-dom/test-utils";
import renderer from "react-test-renderer";
import BillingDetailsForm from "./BillingDetailsForm";
import { Form } from "../../UI";

test("Matches snapshot", () => {
  const wrapper = renderer.create(<BillingDetailsForm />).toJSON();
  expect(wrapper).toMatchSnapshot();
});

test("Calls onSubmit", () => {
  const mockSubmit = jest.fn();
  const wrapper = shallow(<BillingDetailsForm onSubmit={mockSubmit} />);
  const form = wrapper.find(Form);
  act(() => {
    form.simulate("submit");
  });
  expect(mockSubmit).toHaveBeenCalled();
});
