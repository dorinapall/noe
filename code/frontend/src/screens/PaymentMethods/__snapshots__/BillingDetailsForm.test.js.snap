// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<form
  className="Form"
  onSubmit={[Function]}
>
  <div
    className="InputGroup"
  >
    <label
      className="Label "
      htmlFor="company_name"
    >
      Név
    </label>
    <input
      className="Input"
      min={null}
      name="company_name"
      placeholder="Példa Kft."
      type="text"
    />
  </div>
  <div
    className="InputGroup"
  >
    <label
      className="Label "
      htmlFor="country"
    >
      Ország
    </label>
    <input
      className="Input"
      min={null}
      name="country"
      placeholder="Magyarország"
      type="text"
    />
  </div>
  <div
    className="InputGroup"
  >
    <label
      className="Label "
    >
      Tartózkodási cím
    </label>
    <div
      className="InputGroup"
    >
      <div
        className="Flex"
      >
        <div
          style={
            Object {
              "flex": 2,
            }
          }
        >
          <input
            className="Input"
            min={null}
            name="post_code"
            placeholder="Irányítósz."
            type="text"
          />
        </div>
        <div
          style={
            Object {
              "flex": 5,
              "marginLeft": 10,
            }
          }
        >
          <input
            className="Input"
            min={null}
            name="city"
            placeholder="Település"
            type="text"
          />
        </div>
      </div>
    </div>
    <input
      className="Input"
      min={null}
      name="address_line1"
      placeholder="Utca és Házszám"
      type="text"
    />
  </div>
  <div
    className="InputGroup"
  >
    <label
      className="Label "
      htmlFor="tax_number"
    >
      Adószám
    </label>
    <input
      className="Input"
      min={null}
      name="tax_number"
      placeholder="123456789"
      type="text"
    />
  </div>
  <button
    className="Button NextButton"
    type="submit"
  >
    Véglegesítés
  </button>
</form>
`;
